R/dash.R

```diff
@@ -305,12 +305,38 @@ Dash <- R6::R6Class(
           
           # have to format the response body like this
           # https://github.com/plotly/dash/blob/064c811d/dash/dash.py#L562-L584
-          resp <- list(
-            response = list(
-              props = setNames(list(output_value), gsub( "(^.+)(\\.)", "", request$body$output))
+          if (grepl("\\.\\.\\.", request$body$output)) {
+            # if multi-output callback, isolate the output IDs and properties
+            idmatch <- gregexpr("(?<=\\.\\.)(([^\\.]+))(?=\\.)", request$body$output, perl=TRUE)
+            propmatch <- gregexpr("(?<=\\.)(([^\\.]+))(?=\\.\\.)", request$body$output, perl=TRUE)
+            ids <- unlist(regmatches(request$body$output, idmatch))
+            props <- unlist(regmatches(request$body$output, propmatch))
```

Comment:
I'd generally try to avoid regex for simple things like this - my experience in Python and JS is it's slow and prone to errors if not heavily tested.

This could be something like:
```
if (substr(request$body$output, 1, 2) == '..') {
  outputs <- strsplit(substr(request$body$output,3,nchar(s)-2), '...', fixed=TRUE)
  id_props <- sapply(outputs, strsplit, '.', fixed=TRUE)
  ids <- lapply(id_props, '[', 1)
  props <- lapply(id_props, '[', 2)
```

Reply:
Both `grep` and `grepl` are part of base R and written as internal functions using native C code. I don't think performance is an issue here (is it possible to give an example or sample code?), but I'll rewrite this block if it will help avoid errors and unintended behaviour.